src/taipy/gui/data/decimator/scatter_decimator.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import typing as t
+
+import numpy as np
+
+from ..utils import Decimator
+
+
+class ScatterDecimator(Decimator):
+    def __init__(
+        self,
+        binning_ratio: t.Optional[float] = None,
+        max_overlap_points: t.Optional[np.uint8] = None,
+        applied_threshold: t.Optional[int] = None,
```

Comment:
I woudl rename that to 'size_threshold'

Reply:
In this case I prefer applied_theshold to size_threshold since it is the theshold when the decimator will get applied

Reply:
Let me know what you think. Counter me :D